---
title: "DGE_Analysis"
author: "Jonathan Anzules"
output: html_document
date: "2023-06-09"
---

**Introduction**

This file does a couple of things:
 - Prepares all of the relevant metadata data into a table
 - Prepares a transcript to gene mapping of for importing salmon files
 - tximport - imports the salmon quant.sf files into count matrix needed for DESeq
 - deseq - Next we used DESeq2 to do the differential gene expression analysis
 - Results from deseq is used for the Gene Set Enrichment Analysis 

# Package management
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

setwd("/Users/jonan/Documents/Tyseq/Code/") #All file locations will have a relative directionality in the file tree. 
library(tximport) # Importing salmon files and preparing for deseq
library(DESeq2) # DGE Analysis
library(GenomicFeatures) # Generating the tx2gene 
library(ggplot2)
library(tidyr)
library(tidyverse)
library(fgsea) # pathway enrichment analysis
library(biomaRt) # Converting ensembleID to geneID
library(data.table)
library(plotly)
library(dplyr)
```

```{r unloading packages}
# Adding this to test what is making my select() function not work
detach(package:tximport, unload = TRUE)
detach(package:DESeq2, unload = TRUE)
detach(package:GenomicFeatures, unload = TRUE)
detach(package:plotly, unload = TRUE)
detach(package:ggplot2, unload = TRUE)
detach(package:tidyr, unload = TRUE)
detach(package:tidyverse, unload = TRUE)
detach(package:fgsea, unload = TRUE)
detach(package:biomaRt, unload = TRUE)
detach(package:data.table, unload = TRUE)
detach(package:dplyr, unload = TRUE)
```

# Data Prep
```{r tx2gene}
tx2gene <- read.csv("../Data/tx2gene.csv")

# txdb <- makeTxDbFromGFF("../Data/hg19/gencode.v19.chr_patch_hapl_scaff.annotation.gtf")
# 
# # keytypes(txdb)
# k <- keys(txdb, keytype = "TXNAME")
# 
# tx2gene <- AnnotationDbi::select(txdb, k, "GENEID", "TXNAME")
# tx2gene <- as.data.frame(tx2gene)
# tx2gene <- tx2gene[, c("TXNAME", "GENEID")]
# write.csv(tx2gene, file = "../Data/tx2gene.csv", row.names = FALSE)

```
#original
```{r sample setup}
# # Setting up files for upload
# 
# # "../Data/SalmonQuant/quantALL/"
# control_files = c("../Data/SalmonQuant/quantALL/1_XIN460_NHI_ATTACT_L005_R1_001_quant.sf", 
#                   "../Data/SalmonQuant/quantALL/2_XGM061_NHI_TCCGGA_L005_R1_001_quant.sf",
#                   "../Data/SalmonQuant/quantALL/3_XJL334_NHI_CGCTCA_L005_R1_001_quant.sf",
#                   "../Data/SalmonQuant/quantALL/4_ZCA126_NHI_GAGATT_L005_R1_001_quant.sf",
#                   "../Data/SalmonQuant/quantALL/5_XGZ492_NHI_ATTCAG_L005_R1_001_quant.sf")
# 
# T1D_files = c("../Data/SalmonQuant/quantALL/6_AAFS251_T2DHI_GAATTC_L005_R1_001_quant.sf",
#               "../Data/SalmonQuant/quantALL/7_AAJ2482_T2DHI_CTGAAG_L005_R1_001_quant.sf",
#               "../Data/SalmonQuant/quantALL/8_AABW178_T2DHI_TAATGC_L005_R1_001_quant.sf",
#               "../Data/SalmonQuant/quantALL/9_XIX456_T2DHI_CGGCTA_L005_R1_001_quant.sf",
#               "../Data/SalmonQuant/quantALL/10_ABDG032_T2DHI_TCCGCG_L005_R1_001_quant.sf")
# 
# # Creating a sample table that maps each sample to its corresponding quant.sf file
# # _C = control
# # _D = Diabetic
# sample_table <- data.frame(
#   sampleName = c("XIN460_C", "XGM061_C", "XJL334_C", "ZCA126_C", "XGZ492_C",
#                  "AAFS251_D", "AAJ2482_D", "AABW178_D", "XIX456_D", "ABDG032_D"),
#   fileName = c(control_files, T1D_files),
#   diabetes_status = c(rep("No", 5), rep("Yes", 5))
# )

```

# Removed ABDG032_D
```{r}
# Setting up files for upload

# "../Data/SalmonQuant/quantALL/"
control_files = c("../Data/SalmonQuant/quantALL/1_XIN460_NHI_ATTACT_L005_R1_001_quant.sf", 
                  "../Data/SalmonQuant/quantALL/2_XGM061_NHI_TCCGGA_L005_R1_001_quant.sf",
                  "../Data/SalmonQuant/quantALL/3_XJL334_NHI_CGCTCA_L005_R1_001_quant.sf",
                  "../Data/SalmonQuant/quantALL/4_ZCA126_NHI_GAGATT_L005_R1_001_quant.sf",
                  "../Data/SalmonQuant/quantALL/5_XGZ492_NHI_ATTCAG_L005_R1_001_quant.sf")

T1D_files = c("../Data/SalmonQuant/quantALL/6_AAFS251_T2DHI_GAATTC_L005_R1_001_quant.sf",
              "../Data/SalmonQuant/quantALL/7_AAJ2482_T2DHI_CTGAAG_L005_R1_001_quant.sf",
              "../Data/SalmonQuant/quantALL/8_AABW178_T2DHI_TAATGC_L005_R1_001_quant.sf",
              "../Data/SalmonQuant/quantALL/9_XIX456_T2DHI_CGGCTA_L005_R1_001_quant.sf"
              )

# Creating a sample table that maps each sample to its corresponding quant.sf file
# _C = control
# _D = Diabetic
sample_table <- data.frame(
  sampleName = c("XIN460_C", "XGM061_C", "XJL334_C", "ZCA126_C", "XGZ492_C",
                 "AAFS251_D", "AAJ2482_D", "AABW178_D", "XIX456_D"),
  fileName = c(control_files, T1D_files),
  diabetes_status = c(rep("No", 5), rep("Yes", 4))
)
```

```{r}
length(control_files)  # should be 5  
length(T1D_files)      # should be 4  
length(sample_table$sampleName)  # should be 9

```


# DGE Analysis
```{r Loading Count Matrix and Filtering}
# import salmon abundance counts into an r data ckass that can be used by desseq
txi <- tximport(files = sample_table$fileName, type = "salmon", tx2gene = tx2gene, ignoreAfterBar = TRUE) 
counts <- round(txi$counts)
dim(counts)
# Calculate the number of samples in which each gene is expressed for each group
expressed_control <- rowSums(counts[, sample_table$diabetes_status == "No"] > 0)
expressed_diabetes <- rowSums(counts[, sample_table$diabetes_status == "Yes"] > 0)

# Filter genes based on the 80% criterion for each group
keep_genes <- (expressed_control >= floor(0.8 * sum(sample_table$diabetes_status == "No"))) | 
              (expressed_diabetes >= floor(0.8 * sum(sample_table$diabetes_status == "Yes")))

# Apply the filter to the counts matrix
filtered_counts <- counts[keep_genes, ]

# dim(filtered_counts)

```


```{r DESeq differential gene exprl,ession analysis}

columnData <- data.frame(condition = sample_table$diabetes_status)

dds <- DESeqDataSetFromMatrix(countData = filtered_counts, colData = columnData, design = ~condition) # ~condition is what tells it that I am interested in seeing the relationships based on condition

#Adding the right column names to the dds object
colnames(dds) <- sample_table$sampleName
# dds$condition - These should match with colnames(dds); they do.

## Normalizing the counts

# Estimated size factors are determined by the library size
dds <- estimateSizeFactors(dds)

# Differential gene expression analysis
dds <- DESeq(dds)

# Getting the results based on the condition of diabetes or not
results <- results(dds, contrast = c("condition", "Yes", "No"))

# dim(results)
```

```{r - saving normalized counts}

dds <- DESeqDataSetFromMatrix(countData = filtered_counts, colData = columnData, design = ~condition) # ~condition is what tells it that I am interested in seeing the relationships based on condition

#Adding the right column names to the dds object
colnames(dds) <- sample_table$sampleName
# dds$condition - These should match with colnames(dds); they do.

## Normalizing the counts

# Estimated size factors are determined by the library size
dds <- estimateSizeFactors(dds)

# Extract and save normalized counts
norm_counts <- counts(dds, normalized=TRUE)


# Preparing Ensembl IDs (removing version numbers as before)
ensembl_ids <- gsub("\\.\\d+$", "", rownames(norm_counts))
rownames(norm_counts) <- gsub("\\.\\d+$", "", rownames(norm_counts))

# Connecting to Ensembl and retrieve gene names
ensembl <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")

converted_ids <- getBM(attributes = c("ensembl_gene_id", "external_gene_name"),
                       filters = "ensembl_gene_id",
                       values = ensembl_ids,
                       mart = ensembl)

# Create a mapping from Ensembl IDs to Gene Names
id_to_name <- setNames(converted_ids$external_gene_name, converted_ids$ensembl_gene_id)

# Replace row names in norm_counts with gene names, using Ensembl IDs as keys
norm_counts_rownames <- rownames(norm_counts)
rownames(norm_counts) <- ifelse(norm_counts_rownames %in% names(id_to_name), id_to_name[norm_counts_rownames], NA)

dim(norm_counts)

# Filter out rows where row names are NA or empty
norm_counts <- norm_counts[!is.na(rownames(norm_counts)) & rownames(norm_counts) != "", ]

# rownames(norm_counts)
# Save the updated norm_counts data frame

write.csv(norm_counts, "../Data/normalized_gene_expression_data_No_ABDG032.csv", row.names = TRUE)
# write.csv(norm_counts, "../Data/normalized_gene_expression_data.csv", row.names = TRUE)

```

```{r Adding GeneName to results df}
#Rownames to a column
results <- as.data.frame(results)
results$EnsembleID <- rownames(results) 

# Removing the version number from EnsembleID using regex
results$EnsembleID <- gsub("\\.\\d+$", "", results$EnsembleID)

## Obtaining gene ID

# Connect to the Ensembl database using biomaRt
ensembl <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")

# Convert Ensemble IDs to ensembl_gene_id Gene IDs
converted_ids <- getBM(attributes = c("ensembl_gene_id", "external_gene_name"),
                       filters = "ensembl_gene_id",
                       values = results$EnsembleID,
                       mart = ensembl)

# Merge the converted IDs with gene expression data
results <- merge(results, converted_ids, by.x = "EnsembleID", by.y = "ensembl_gene_id")
  #the dimensions of results went from 15295 x 7 to 14962 x 8
# dim(results)

# Removing empty and NA genesymbol entries
results <- subset(results, !(is.na(external_gene_name) | external_gene_name == ""))
# dim(results)

#Renaming column
colnames(results)[colnames(results) == "external_gene_name"] <- "GeneName"

#Reordering the columns
results <- results[, c("GeneName", setdiff(names(results), "GeneName"))]

# Calculate the negative log10 of the p-values for plotting
results$neg_log10_pvalue <- -log10(results$pvalue)
results$neg_log10_Adjpvalue <- -log10(results$padj)


```


```{r Saving results}


# write.csv(results, "../Data/ForReview/DESeqResults_No_ABDG032.csv", row.names = FALSE)
# write.csv(results, "../Data/ForReview/DESeqResults.csv", row.names = FALSE)

# Filtering for significant genes based on pvalue
sigGenes_p <- subset(results, pvalue <= 0.05)
sigGenes_p <- sigGenes_p[order(sigGenes_p$pvalue),]

# Filtering for significant genes based on adjusted pvalue
sigGenes_padj <- subset(results, padj <= 0.1)
sigGenes_padj <- sigGenes_padj[order(sigGenes_padj$pvalue),]

dim(sigGenes_p)
dim(sigGenes_padj)

# Saving all df's
# write.csv(results, "../Data/DGE_Results.csv", row.names = FALSE)
# write.csv(sigGenes_p, file = "../Data/DGE_sig_genes_p.csv", row.names = FALSE)
# write.csv(sigGenes_padj, file = "../Data/DGE_sig_genes_adjp.csv", row.names = FALSE)

write.csv(results, "../Data/DGE_Results_No_ABDG032.csv", row.names = FALSE)
write.csv(sigGenes_p, file = "../Data/DGE_sig_genes_p_No_ABDG032.csv", row.names = FALSE)
write.csv(sigGenes_padj, file = "../Data/DGE_sig_genes_adjp_No_ABDG032.csv", row.names = FALSE)

```


```{r - DESeqDataSet to tabular data: For Madelyn}

# Extracting expression values from DESeqDataSet
expression_values <- assay(dds, "counts")

# Creating a dataframe
gene_expression_data <- as.data.frame(expression_values)
gene_expression_data$EnsembleID <- rownames(gene_expression_data)

# Removing the version number from EnsembleID using regex
gene_expression_data$EnsembleID <- gsub("\\.\\d+$", "", gene_expression_data$EnsembleID)

## Obtaining gene ID

# Connect to the Ensembl database using biomaRt
ensembl <- useMart("ensembl", dataset = "hsapiens_gene_ensembl")

# Convert Ensemble IDs to ensembl_gene_id Gene IDs
converted_ids <- getBM(attributes = c("ensembl_gene_id", "external_gene_name"),
                       filters = "ensembl_gene_id",
                       values = gene_expression_data$EnsembleID,
                       mart = ensembl)

# Merge the converted IDs with your gene expression data
gene_expression_data <- merge(gene_expression_data, converted_ids, by.x = "EnsembleID", by.y = "ensembl_gene_id")

# Removing empty and NA genesymbol entries
gene_expression_data <- subset(gene_expression_data, !(is.na(external_gene_name) | external_gene_name == ""))

# write.csv(gene_expression_data, "../Data/ExpressionData_filtered.csv")
write.csv(gene_expression_data, "../Data/ExpressionData_filtered_No_ABDG032.csv")

```


```{r Sanity check}
# Checking to see that the right columns are associated to the right patient ID

# Get the column names from DESeq object
col_names = colnames(dds)

# Get the patient IDs from the sample table
patient_ids <- sample_table$sampleName

# Compare the column names with the sample name
# If match 
match_result <- match(col_names, patient_ids)

match_result

# Print the results
for (i in seq_along(col_names)) {
  cat("Column:", col_names[i], " - SampleName ID:", patient_ids[match_result[i]], "\n")
}

```

# Exploring Results

```{r top10}
# #Normalizing the counts

top_genes <- results[order(results$padj), ]
top_10_genes <- head(top_genes, n=10)

top_10_genes

# write.csv(top_10_genes, file = "../Data/to_10_genes.csv")


```

```{r top30}

top_genes <- results[order(results$padj), ]
top_30_genes <- head(top_genes, n=30)

top_30_genes

write.csv(top_30_genes, file = "../Data/top_30_genes_2.csv")
```

```{r p-values vs log2foldchange}
# Checking to see the what differences is if I sort by smallest p-values and the
# greatest absolute value in log2foldchange

results

# Sort results by absolute log2FC in descending order
sorted_results <- results[order(abs(results$log2FoldChange), decreasing = TRUE), ]

# Print the sorted results
print(sorted_results)

```

```{r - Identifying Differences}
# I updated salmon to the most up to date version and used the gcbias flag - _2

library(dplyr)
dfA = read.csv("../Data/top_30_genes.csv")
dfB = read.csv("../Data/top_30_genes_2.csv")


# Find values in dfA that are not present in dfB
values_only_in_dfA <- anti_join(dfA, dfB, by = "EnsembleID")

# Find values in dfB that are not present in dfA
values_only_in_dfB <- anti_join(dfB, dfA, by = "EnsembleID")

values_only_in_dfA
values_only_in_dfB

```
#Volcano plots

```{r - Loading Data}
# Results  from DGE analysis
results = read.csv("../Data/DGE_Results.csv")

```


```{r - Adj p value Volcano plot with selected labels}

#Subset top up/down regulated genes
Upregulated_Genes <- subset(results, log2FoldChange > 0 )
Downregualted_Genes <- subset(results, log2FoldChange < 0)

# Choosing the number of top/down
# Number_of_TopnDown = 6


# Select the top 10 upregulated and top 10 downregulated genes
top_upregulated <- Upregulated_Genes[order(Upregulated_Genes$padj, na.last = NA)[1:10],]
top_downregulated <- Downregualted_Genes[order(Downregualted_Genes$padj, na.last = NA)[1:1],]

# Combine the two dataframes
selection <- rbind(top_upregulated, top_downregulated)

# Create a volcano plot
volcano_plot <-  ggplot(results, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(color = "grey", alpha = 0.6) +
  geom_point(data = subset(results, padj < 0.05 & abs(log2FoldChange) > 1),
             color = "red", alpha = 0.6) +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "blue") +
  # Adding the text labels for the top genes
  geom_text(data = selection, aes(label = GeneName), 
            vjust = -1, hjust = 1) +
  labs(x = "log2 Fold Change", y = "-log10(adjusted p-value)",
       title = "Volcano Plot") +
  theme_bw()

ggsave(filename = "../Figures/volcano_adj_labeled.png", plot = volcano_plot,
       width = 10, dpi = 300)


```

```{r - Volcano plot with selected labels}

#Subset top up/down regulated genes
Upregulated_Genes <- subset(results, log2FoldChange > 0 )
Downregualted_Genes <- subset(results, log2FoldChange < 0)

# Choosing the number of top/down
Number_of_TopnDown = 6


# Select the top 10 upregulated and top 10 downregulated genes
top_upregulated <- Upregulated_Genes[order(Upregulated_Genes$pvalue, na.last = NA)[1:Number_of_TopnDown],]
top_downregulated <- Downregualted_Genes[order(Downregualted_Genes$pvalue, na.last = NA)[1:Number_of_TopnDown],]

# Combine the two dataframes
selection <- rbind(top_upregulated, top_downregulated)

# Create a volcano plot
volcano_plot <- ggplot(results, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(color = "grey", alpha = 0.6) +
  geom_point(data = subset(results, pvalue < 0.05 & abs(log2FoldChange) > 1),
             color = "red", alpha = 0.6) +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "blue") +
  # Adding the text labels for the top genes
  geom_text(data = selection, aes(label = GeneName), 
            vjust = -1, hjust = 1) +
  labs(x = "log2 Fold Change", y = "-log10(adjusted p-value)",
       title = "Volcano Plot") +
  theme_bw()

ggsave(filename = "../Data/ForReview/volcano_top_n_down.png", plot = volcano_plot,
       width = 10, dpi = 300)
# write.csv(results, "../Data/ForReview/DESeqResults.csv")
```


```{r -  Plot No label - adjusted p value}
# Results generated in Data Prep
results_df <- as.data.frame(results)

# Create a volcano plot
ggplot(results_df, aes(x = log2FoldChange, y = neg_log10_padj)) +
  geom_point(color = "grey", alpha = 0.6) +
  geom_point(data = subset(results_df, padj < 0.05 & abs(log2FoldChange) > 1),
             color = "red", alpha = 0.6) +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "blue") +
  labs(x = "log2 Fold Change", y = "-log10(adjusted p-value)",
       title = "Volcano Plot") +
  theme_bw()

top_genes <- results_df[order(results_df$padj), ]
top_10_genes <- head(top_genes, n=10)

write.csv(top_genes, "../Data/topDGEGenes_filtered.csv")

```

```{r -  Plot No label - p value }

# Create a volcano plot
ggplot(results_df, aes(x = log2FoldChange, y = neg_log10_pvalue)) +
  geom_point(color = "grey", alpha = 0.6) +
  geom_point(data = subset(results_df, pvalue < 0.05 & abs(log2FoldChange) > 1),
             color = "red", alpha = 0.6) +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "blue") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "blue") +
  labs(x = "log2 Fold Change", y = "-log10(p-value)",
       title = "Volcano Plot") +
  theme_bw()
```

## html volcano plots
```{r html file}

# Convert 'padj' column to numeric
results$padj <- as.numeric(results$padj)

# Create the interactive volcano plot with hover text
volcano_plot <- plot_ly(data = results, 
                        x = ~log2FoldChange, 
                        y = -log10(results$padj), # No ~ symbol for log10() function
                        text = ~EnsembleID,
                        hoverinfo = "text",
                        type = "scatter",
                        mode = "markers",
                        marker = list(
                          size = 10,
                          color = ifelse(results_df$padj < 0.05, "red", "blue"),
                          opacity = 0.7
                        )) %>%
  layout(title = "Volcano Plot",
         xaxis = list(title = "log2 Fold Change"),
         yaxis = list(title = "-log10 p-value"))

# Save the interactive plot as an HTML file
htmlwidgets::saveWidget(volcano_plot, file = "../Figures/interactive_volcano_plot.html")


```

```{r html figure}
# Load necessary libraries
library(plotly)
library(dplyr)

# Adjusted p-value cutoff for highlighting most differentially expressed genes
pvalue_cutoff <- 0.05

# Create a new column to indicate significance based on adjusted p-value
results$significant <- ifelse(results$padj <= pvalue_cutoff, "Yes", "No")

# Create the volcano plot with interactive tooltips for Ensemble ID
volcano_plot <- plot_ly(data = results, x = ~log2FoldChange, y = ~-log10(padj), 
                        color = ~significant, colors = list("Yes" = "red", "No" = "black")) %>%
  add_markers(size = 5, alpha = 0.6, hoverinfo = "text",
              text = ~paste("Ensemble ID: ", EnsembleID, "<br>",
                            "log2 Fold Change: ", log2FoldChange, "<br>",
                            "Adjusted p-value: ", padj)) %>%
  layout(xaxis_title = "log2 Fold Change",
         yaxis_title = "-log10(Adjusted p-value)",
         title = "Volcano Plot",
         hovermode = "closest")


plot_ly(data = results, x = ~log2FoldChange, y = ~-log10(padj), 
                        color = ~significant, colors = list("Yes" = "red", "No" = "black")) %>%
  add_markers(size = 5, alpha = 0.6, hoverinfo = "text",
              text = ~paste("Ensemble ID: ", EnsembleID, "<br>",
                            "log2 Fold Change: ", log2FoldChange, "<br>",
                            "Adjusted p-value: ", padj)) %>%
  layout(xaxis_title = "log2 Fold Change",
         yaxis_title = "-log10(Adjusted p-value)",
         title = "Volcano Plot",
         hovermode = "closest")


# Convert plot_ly object to plotly object
plotly_obj <- plotly_build(volcano_plot)

# Convert to JSON format
plotly_json <- plotly:::plotly_json(plotly_obj, FALSE)

# Save the plot as an interactive HTML file
output_file <- "../Figures/interactive_volcano_plot2.html"
htmlwidgets::saveWidget(list(data = plotly_json), file = output_file)


```

```{r}
# Adjusted p-value cutoff for highlighting most differentially expressed genes
pvalue_cutoff <- 0.05

# Create a new column to indicate significance based on adjusted p-value
results$significant <- ifelse(results$padj < pvalue_cutoff, "Yes", "No")

# Create the volcano plot with interactive tooltips for Ensemble ID
# There is something wrong with the color selection
volcano_plot <- plot_ly(data = results, x = ~log2FoldChange, y = ~-log10(padj), 
                        color = ~significant, colors = list("Yes" = "red", "No" = "black")) %>%
  add_markers(size = 5, alpha = 0.6, hoverinfo = "text",
              text = ~paste("Ensemble ID: ", EnsembleID, "<br>",
                            "log2 Fold Change: ", log2FoldChange, "<br>",
                            "Adjusted p-value: ", padj)) %>%
  layout(xaxis_title = "log2 Fold Change",
         yaxis_title = "-log10(Adjusted p-value)",
         title = "Volcano Plot",
         hovermode = "closest")

# Save the plot as an interactive HTML file
output_file <- "../Figures/interactive_volcano_plot2.html"
htmlwidgets::saveWidget(volcano_plot, file = output_file)

```


# Exploring the results
```{r}
results <- read.csv("../Data/DGE_Results.csv")

sigResults = subset(results, pvalue < 0.05)
PsigResults = subset(results, padj < 0.1)
dim(results)
dim(sigResults)
dim(PsigResults)
```




















